grakn-bootup/src/main/java/ai/grakn/bootup/StorageConfig.java

```
@@ -0,0 +1,118 @@
+
+    private final ImmutableMap<String, Object> yamlParams;
+    private static final String EMPTY_VALUE = "";
+    private static final String CONFIG_PARAM_PREFIX = "storage.internal.";
+    private static final String DEFAULT_DATA_DIR = "db/";
+    private static final String SAVED_CACHES_SUBDIR = "cassandra/saved_caches";
+    private static final String COMMITLOG_SUBDIR = "cassandra/commitlog";
+    private static final String DATA_SUBDIR = "cassandra/data";
+
+    private StorageConfig(Map<String, Object> yamlParams){
+        this.yamlParams = ImmutableMap.copyOf(yamlParams);
+    }
+
+    private static Map<String, Object> parseStringToMap(String yaml){
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.MINIMIZE_QUOTES));
+        try {
+            TypeReference<Map<String, Object>> reference = new TypeReference<Map<String, Object>>(){};
+            Map<String, Object> yamlParams = mapper.readValue(yaml, reference);
+            return yamlParams.entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue() == null ? EMPTY_VALUE : e.getValue()));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static StorageConfig of(String yaml) {
+        return new StorageConfig(parseStringToMap(yaml));
+    }
+
+    public String toYamlString() {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.MINIMIZE_QUOTES));
+        try {
+            ByteArrayOutputStream outputstream = new ByteArrayOutputStream();
+            mapper.writeValue(outputstream, yamlParams);
+            return outputstream.toString(StandardCharsets.UTF_8.name());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private StorageConfig updateDataDirs(GraknConfig config) {
+        String dbDir = config.properties().containsKey(GraknConfigKey.DATA_DIR)?
```

In my opinion, we should avoid having this logic of falling back to some default value. We should just let the program crash if the user mistakenly delete the `db-dir` property.

While this behavior is nice, it can be a little inconsistent since the user may then expect that other parameters will fall back to some default value if removed as well. And that might not always be the case.